src/main/java/org/jetbrains/nativecerts/mac/SecurityFrameworkUtil.java

```diff
@@ -118,6 +125,34 @@ static boolean isSelfSignedCertificate(X509Certificate certificate) {
         return true;
     }
 
+    private static boolean validateCertificate(SecurityFramework.SecCertificateRef certificateRef) {
+        SecurityFramework.SecTrustRef policy = null;
+        try {
+            final Pointer[] values = {certificateRef.getPointer()};
+            CoreFoundation.CFArrayRef subjCerts = CoreFoundationExt.INSTANCE.CFArrayCreate(
+                    null, values, new CoreFoundation.CFIndex(1), null
+            );
+
+            SecurityFramework.SecTrustRefByReference secTrustRefByReference = new SecurityFramework.SecTrustRefByReference();
+
+            policy = SecurityFramework.INSTANCE.SecPolicyCreateBasicX509();
```

Comment:
Ideally I'll change this to an SSL policy rather than just the basic x509, this one is easier as there's a built-in function for it.